wp-includes/js/service-worker.js

```diff
@@ -148,3 +148,10 @@ self.addEventListener( 'fetch', event => {
 		event.respondWith( responsePromise );
 	}
 } );
+
+// Skip the waiting phase for the Service Worker.
+self.addEventListener( 'message', function ( event ) {
+	if ( 'skipWaiting' === event.data.action ) {
```

Comment:
Is sending a message the right way to do this? Should it not be `self.skipWaiting` unconditionally?

For example, in PHP-generated JS that is added to the service worker:

```php
if ( apply_filters( 'wp_service_worker_skip_waiting', true ) ) {
    echo 'self.skipWaiting();';
}
```

I see this similar to how we're managing navigation preload.

Reply:
Changed this within f1ca81a.